changelogs/unreleased/build-clang-fails-because-of-null-pointer-subtraction.md

```diff
@@ -0,0 +1,3 @@
+## bugfix/build
```

Comment:
We do have a `build` subsystem, but AFAI, it only includes CMake and CI/CD scripts.

The subsystem you patch here is simply called `sql`. Same comment regarding your commit message subject.

Nit: your commit message is quite formal — please provide some context, e.g.:

```git
sql: fix -Wnull-pointer-subtraction warning

clang 13 includes a check for subtraction from `NULL` pointer which is considered UB: historically, we had an alignment checking macro which is affected by this. It seems like the intention of the macro's author was to implicitly cast the pointer being checked to `uintptr_t` without including `stddef.h` — replace this subtraction with an explicit cast.
```